libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts

```diff
@@ -299,7 +333,14 @@ export class UpdateWorkflow {
       );
     }
 
-    return notificationTemplateWithStepTemplate;
+    return this.getWorkflowByIdsUseCase.execute(
+      GetWorkflowByIdsCommand.create({
+        userId: command.userId,
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        identifierOrInternalId: command.id,
+      }),
+    );
```

Comment:
It's okay for now. One thing I noticed, don't we also fetch at [this point](https://github.com/novuhq/novu/pull/7005/files#diff-215a725ccdc52f5207a653ad21479e039d533723daef8101e1897e14b8da5d30R302)? If not mistaken, no updates take place since that fetching.

Reply:
We can also use https://mongoosejs.com/docs/tutorials/findoneandupdate.html and do everything in a single query.